runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -172,6 +173,41 @@ J9::ARM64::TreeEvaluator::checkcastEvaluator(TR::Node *node, TR::CodeGenerator *
 
    return targetRegister;
    }
+	
+TR::Register *
+J9::ARM64::TreeEvaluator::flushEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::ILOpCodes op = node->getOpCodeValue();
+
+   if (op == TR::allocationFence)
+      {
+      if (!node->canOmitSync())
+         {
+         // Data synchronization barrier -- 0xF to turn on option for both reads and writes
+         generateSynchronizationInstruction(cg, TR::InstOpCode::dsb, node, 0xF);
+         }
+      }
+   else
+      {
+      if (op == TR::loadFence)
+         {
+         // Data synchronization barrier -- 0xD to turn on option for reads
+         generateSynchronizationInstruction(cg, TR::InstOpCode::dsb, node, 0xD);
+         }
+      else if (op == TR::storeFence)
+         {
+         // Data synchronization barrier -- 0xE to turn on option for writes
+         generateSynchronizationInstruction(cg, TR::InstOpCode::dsb, node, 0xE);
+         }
+      else
+         {
+         // Data synchronization barrier -- 0xF to turn on option for both reads and writes
+         generateSynchronizationInstruction(cg, TR::InstOpCode::dsb, node, 0xF);
+         }
```

Comment:
Could you modify this if/else ladder so that you only have one call to `generateSynchronizationInstruction` please?  It will save some space in the .so.